src/ast/passes/codegen_llvm.cpp

```diff
@@ -327,10 +327,37 @@ void CodegenLLVM::visit(Builtin &builtin)
     b_.CreateGetCurrentComm(ctx_, buf, builtin.type.GetSize(), builtin.loc);
     expr_ = buf;
     expr_deleter_ = [this, buf]() { b_.CreateLifetimeEnd(buf); };
+  } else if (builtin.ident == "func") {
+    // kfunc/kretfunc probes do not have access to registers, so require use of
+    // the get_func_ip helper to get the instruction pointer.
+    //
+    // For [ku]retprobes, the IP register will not be pointing to the function
+    // we want to trace. It may point to a kernel trampoline, or it may point to
+    // the caller of the traced function, as it fires after the "ret"
+    // instruction has executed.
+    //
+    // The get_func_ip helper resolves these issues for us.
+    //
+    // But do not use the it for non-ret [ku]probes, as the helper will fail for
```

Comment:
Curious, do you mean if folks pass an offset or if the function specified has been inlined?